core/http-auth-aws/src/main/java/software/amazon/awssdk/http/auth/aws/internal/signer/chunkedencoding/ChunkedEncodedPublisher.java

```diff
@@ -174,6 +201,41 @@ private int calculateExtensionsLength(List<Pair<byte[], byte[]>> chunkExtensions
                        }).sum();
     }
 
+    private List<ByteBuffer> getTrailerData() {
+        List<ByteBuffer> data = new ArrayList<>();
+
+        for (TrailerProvider provider : trailers) {
```

Comment:
just query : in case of Checksum trailers we will have only Trailers for each chunks right ? That is trailers size will be only 1 right ? In which cases will trailers have greater than 1

Reply:
In general chunk encoded bodies support any number of trailers, see https://datatracker.ietf.org/doc/html/rfc7230#section-4.1.2

```
  trailer-part   = *( header-field CRLF )
```

This support is also in the `InputStream` version: https://github.com/aws/aws-sdk-java-v2/blob/c6ee51b51d4925c574127327507341e97a9cdc89/core/http-auth-aws/src/main/java/software/amazon/awssdk/http/auth/aws/internal/signer/chunkedencoding/ChunkedEncodedInputStream.java#L274